src/HassClient/Client/HassClient.cs

```diff
@@ -319,6 +357,17 @@ public class HassClient : IHassClient
             _pipelineFactory = pipelineFactory;
             _wsFactory = wsFactory;
             _httpClient = httpClient;
+
+            HassEventsObservable = Observable.FromEventPattern<EventHandler<HassEvent>, HassEvent>(
+                a => HassEvents += a,
+                a => HassEvents -= a).Select(e => e.EventArgs)
+                .AsConcurrent(t => TrackBackgroundTask(t));

```

Comment:
Maybe the AsConcurrent() with tracking of Background tasks should not be done by HassClient. But just provide a serial stream of all events in the same order as they come in. If a client wants to process them concurrent they can, but sometimes you might not want that.

Reply:
Good point!

Reply:
I changed that to your suggested method. Thanks.